src/Sentry/Internal/Extensions/JsonExtensions.cs

```diff
@@ -415,13 +463,46 @@ public static void Deconstruct(this JsonProperty jsonProperty, out string name,
         {
             writer.WriteStringValue(dto);
         }
+        else if (value is TimeSpan timeSpan)
+        {
+            writer.WriteStringValue(timeSpan.ToString("g", CultureInfo.InvariantCulture));
+        }
+#if NET6_0_OR_GREATER
+        else if (value is DateOnly date)
+        {
+            writer.WriteStringValue(date.ToString("O", CultureInfo.InvariantCulture));
+        }
+        else if (value is TimeOnly time)
+        {
+            writer.WriteStringValue(time.ToString("HH:mm:ss.FFFFFFF", CultureInfo.InvariantCulture));
```

Comment:
I won't pretend I know what I'm talking about, but I have a question - `"FFFFFFF"` will not preserve zeros if the trailing digits are zero. `"ffffff"` will do. Is the intent here to ignore the trailing zeroes for the serialized value?

Reply:
Yes. The intent is not serialize extra zeros, so F is appropriate.